FrejaEidClient/src/main/java/com/verisec/frejaeid/client/client/impl/AuthenticationClient.java

```diff
@@ -49,55 +53,72 @@ public static Builder create(SslSettings sslSettings, FrejaEnvironment frejaEnvi
         requestValidationService.validateInitAuthRequest(initiateAuthenticationRequest, authenticationService.getTransactionContext());
-        return authenticationService.initiate(initiateAuthenticationRequest).getAuthRef();
+        LOG.debug("Initiating authentication transaction for user info type {}, minimum registration level of user {} and requesting attributes {}.", initiateAuthenticationRequest.getUserInfoType(), initiateAuthenticationRequest.getMinRegistrationLevel().getState(), initiateAuthenticationRequest.getAttributesToReturn());
+        String reference = authenticationService.initiate(initiateAuthenticationRequest).getAuthRef();
+        LOG.debug("Received authetnication transaction reference {}.", reference);
+        return reference;
     }
-
+    
     @Override
     public AuthenticationResult getResult(AuthenticationResultRequest authenticationResultRequest) throws FrejaEidClientInternalException, FrejaEidException {
         requestValidationService.validateResultRequest(authenticationResultRequest);
-        return authenticationService.getResult(authenticationResultRequest);
+        LOG.debug("Getting result for authentication transaction reference {}.", authenticationResultRequest.getAuthRef());
+        AuthenticationResult authenticationResult = authenticationService.getResult(authenticationResultRequest);
+        LOG.debug("Received {} status for authentication transaction reference {}.", authenticationResult.getStatus(), authenticationResult.getAuthRef());
+        return authenticationResult;
     }
-
+    
     @Override
     public List<AuthenticationResult> getResults(AuthenticationResultsRequest authenticationResultsRequest) throws FrejaEidClientInternalException, FrejaEidException {
         requestValidationService.validateResultsRequest(authenticationResultsRequest);
-        return authenticationService.getResults(authenticationResultsRequest).getAuthenticationResults();
+        LOG.debug("Getting all authentication transaction results.");
+        List<AuthenticationResult> authenticationResults = authenticationService.getResults(authenticationResultsRequest).getAuthenticationResults();
+        LOG.debug("Successfully received authentication transaction results.");
+        return authenticationResults;
     }
-
+    
     @Override
     public AuthenticationResult pollForResult(AuthenticationResultRequest authenticationResultRequest, int maxWaitingTimeInSec) throws FrejaEidClientInternalException, FrejaEidException, FrejaEidClientPollingException {
         requestValidationService.validateResultRequest(authenticationResultRequest);
-        return authenticationService.pollForResult(authenticationResultRequest, maxWaitingTimeInSec);
+        LOG.debug("Polling {}s for result for authentication transaction reference {}.", maxWaitingTimeInSec, authenticationResultRequest.getAuthRef());
```

Comment:
I think the log message would be a bit clearer if you say "seconds" instead of using the abbreviation.